middle_end/flambda/simplify/simplify_expr.rec.ml

```diff
@@ -732,9 +732,9 @@ and simplify_direct_tuple_application
     -> 'a k -> Expr.t * 'a * UA.t
 = fun dacc apply code_id k ->
   let dbg = Apply.dbg apply in
-  let callee's_params_and_body = DE.find_code (DA.denv dacc) code_id in
+  let callee's_code = DE.find_code (DA.denv dacc) code_id in
   let param_arity =
-    Function_params_and_body.params_arity callee's_params_and_body
+    Code.params_arity callee's_code
```

Comment:
I thought it was `Function_params_and_arity.t` that always had the full arity and it's the closures that were sometimes tupled. I'm a little troubled that I can't find any instances of `is_tupled:true` or `is_tupled = true`, though.

As I recall, I had to rebase over a change that removed the code that was creating tupled stubs, so it's very possible I incorrectly merged something.

Reply:
I've realised what this is: the original code called `Function_params_and_body.params_arity` rather than using an accessor on `Function_declaration`.  Nothing wrong.